### ISSUE TITLE
Parse error with valid syntax: Cannot convert 0x0 O to a BigInt

### ISSUE THREAD

**Issue by flit** [microsoft/pyright#3569]
**Describe the bug**

Internal error exception while parsing the valid [pyocd/gdbserver/syscall.py](https://github.com/pyocd/pyOCD/blob/main/pyocd/gdbserver/syscall.py) file from the [pyocd](https://github.com/pyocd/pyocd) project.

```
An internal error occurred while parsing file "/Users/creed/projects/pyocd/pyocd/gdbserver/syscall.py": SyntaxError: Cannot convert 0x0
O to a BigInt
    at BigInt (<anonymous>)
    at t.Tokenizer._tryNumber (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/parser/tokenizer.ts:794:41)
    at t.Tokenizer._handleCharacter (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/parser/tokenizer.ts:494:30)
    at t.Tokenizer._addNextToken (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/parser/tokenizer.ts:349:19)
    at t.Tokenizer.tokenize (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/parser/tokenizer.ts:264:18)
    at v._startNewParse (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/parser/parser.ts:332:43)
    at callback (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/parser/parser.ts:223:18)
    at s.timeOperation (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/common/timing.ts:44:28)
    at v.parseSourceFile (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/parser/parser.ts:222:38)
    at callback (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/sourceFile.ts:761:45)
    at t.LogTracker.log (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/common/logTracker.ts:43:20)
    at t.SourceFile.parse (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/sourceFile.ts:712:33)
    at R._parseFile (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/program.ts:863:36)
    at R._bindFile (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/program.ts:888:14)
    at callback (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/program.ts:966:22)
    at subtractFromTime (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/common/timing.ts:60:13)
    at importLookup (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/program.ts:965:41)
    at Object.resolveAliasDeclaration (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/aliasDeclarationUtils.ts:62:28)
    at Object.resolveAliasDeclaration (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/typeEvaluator.ts:18719:33)
    at callback (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/typeEvaluatorWithTracker.ts:95:64)
    at s.timeOperation (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/common/timing.ts:44:28)
    at run (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/typeEvaluatorWithTracker.ts:43:46)
    at Object.resolveAliasDeclaration (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/typeEvaluatorWithTracker.ts:95:13)
    at _0x413c36._getNameNodeToken (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/server.bundle.js:1:287523)
    at _0x413c36.visitName (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/server.bundle.js:1:283966)
    at _0x413c36.visitNode (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:255:29)
    at _0x413c36.walk (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:95:37)
    at forEach (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:104:22)
    at Array.forEach (<anonymous>)
    at _0x413c36.walkMultiple (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:102:15)
    at _0x413c36.walk (/Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:97:18)
    at /Users/creed/.vscode/extensions/ms-python.vscode-pylance-2022.6.10/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:104:22
```

**To Reproduce**

Open https://github.com/pyocd/pyOCD/blob/main/pyocd/gdbserver/syscall.py in VSCode with pyright/pylance enabled. Alternatively, run `pyright` on this file.

For your convenience, the offending file is attached: [syscall.py.zip](https://github.com/microsoft/pyright/files/8884723/syscall.py.zip)

**Expected behavior**

No exception. 😉 

**Code**

The problem seems to be related to this section at the top of the file:
```python
O_RDONLY = 0x0
O_WRONLY = 0x1
O_RDWR = 0x2
O_APPEND = 0x8
O_CREAT = 0x200
O_TRUNC = 0x400
O_EXCL = 0x800
```

Removing the entire section resolves the issue. Removing only one (any one) or even a few lines doesn't predictably resolve the error.

**VS Code extension or command-line**
Pyright is running as VS Code Pylance v2022.6.10. Also reproduced with the VSCode Pyright extension and command line pyright v1.1.253.

**Comment by erictraut** [comment]
Thanks for the bug report. This is a bug in the tokenizer logic. It is triggered only in the case where a hex literal token is followed by a newline followed by a lowercase or uppercase 'o' character.

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.254, which I just published. It will also be included in the next release of pylance.

**Comment by flit** [comment]
Thanks!
